Foundation/NSXMLDocument.swift

```diff
@@ -53,147 +53,273 @@ public enum NSXMLDocumentContentKind : UInt {
 	@discussion Note: if the application of a method would result in more than one element in the children array, an exception is thrown. Trying to add a document, namespace, attribute, or node with a parent also throws an exception. To add a node with a parent first detach or create a copy of it.
 */
 public class NSXMLDocument : NSXMLNode {
-    
+    private var _xmlDoc: _CFXMLDocPtr {
+        get {
+            return _CFXMLDocPtr(_xmlNode)
+        }
+    }
     /*!
         @method initWithXMLString:options:error:
         @abstract Returns a document created from either XML or HTML, if the HTMLTidy option is set. Parse errors are returned in <tt>error</tt>.
     */
-    public convenience init(XMLString string: String, options mask: Int) throws { NSUnimplemented() }
-    
+    public convenience init(XMLString string: String, options mask: Int) throws {
+        let data: NSData = string.withCString {
```

Comment:
So in trying to use this function, I believe I have found a bug in the conversion of Swift strings to CFStrings. `CFStringCreateExternalRepresentation` calls `__CFStrLength` which in turn uses `__CFStrIsInline` to determine where to find the length. Unfortunately, that function is returning `true` for a situation where it obviously isn't true! This leads to an erroneous length being returned, and obviously we get a crash as a result.